### PR TITLE
568 married transferred ssn

### PR DESCRIPTION
**Pull request by smacadam**
Ticket 568, Marital Status, Married During Pregnancy, Paternity Acknowledgement, Mother Transferred, Infant Living, Infant Transferred, Number of Live Born, and SSN Requested for Child.

Refactored the Observation helpers to allow us to get and create observations with different value types. Moved Marc's initial helper functions to VitalRecord_util.cs

### CODE REVIEW

File: projects/BFDR.Tests/BirthRecord_Should.cs [hunk 1]
```diff
@@ -1539,5 +1539,157 @@ public void SetRegistrationDate()
       Assert.Null(SetterBirthRecord.RegistrationDateDay);
     }
 
+    public void SetMaritalStatus()
```
Comment (RobertScalfani) [inline projects/BFDR.Tests/BirthRecord_Should.cs:1542]: Add [Fact]

File: projects/BFDR/BirthRecord_submissionProperties.cs [hunk 2]
```diff
@@ -5061,7 +5032,7 @@ public string FirstPrenatalCareVisit
                 {
                     return (obs.Value as Hl7.Fhir.Model.Date)?.Value;
                 }
-                return null;
+                                return null;
```
Comment (RobertScalfani) [inline projects/BFDR/BirthRecord_submissionProperties.cs:5035]: Formatting

**Review by RobertScalfani — CHANGES_REQUESTED** [review]
Nitpicks, but looks like it covered everything that I could find. And there are conflicts to resolve now.

**Review by RobertScalfani — APPROVED** [review]

File: projects/VitalRecord/VitalRecord_util.cs [hunk 3]
```diff
@@ -94,30 +94,19 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
             }
         }
 
-        // TODO: How can we make this flexible to support more types?
+
         /// <summary>Helper to support vital record property getter helper methods for values stored in Observations.</summary>
         /// <param name="code">the code to identify the type of Observation</param>
         /// <param name="extensionURL">if present, specifies that the value should be get from an extension with the provided URL instead</param>
```
Comment (pkmitre) [inline projects/VitalRecord/VitalRecord_util.cs:100]: Would remove the reference to "extensionURL" param for this method.

File: projects/VitalRecord/VitalRecord_util.cs [hunk 4]
```diff
@@ -129,7 +118,7 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
         /// <param name="section">the section of the composition the Observation should be added to</param>
         /// <param name="extensionURL">if present, specifies that the value should be set on an extension with the provided URL instead</param>
         /// <param name="propertyName">the name of the C# property, used to determine the subject ID</param>
-        protected void SetObservationValue(Dictionary<string, string> value, string code, string codeSystem, string text, string profileURL, string section, string extensionURL = null, [CallerMemberName] string propertyName = null)
+        protected Observation CreateObservation(string code, string codeSystem, string text, string profileURL, string section, string focusId = null, [CallerMemberName] string propertyName = null)
```
Comment (pkmitre) [inline projects/VitalRecord/VitalRecord_util.cs:121]: Should add focusId to the param list above.

File: projects/VitalRecord/VitalRecord_util.cs [hunk 5]
```diff
@@ -94,30 +94,19 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
             }
         }
 
-        // TODO: How can we make this flexible to support more types?
+
         /// <summary>Helper to support vital record property getter helper methods for values stored in Observations.</summary>
         /// <param name="code">the code to identify the type of Observation</param>
         /// <param name="extensionURL">if present, specifies that the value should be get from an extension with the provided URL instead</param>
-        protected Dictionary<string, string> GetObservationValue(string code, string extensionURL = null)
+        protected Observation GetObservation(string code)
```
Comment (pkmitre) [inline projects/VitalRecord/VitalRecord_util.cs:101]: Seems like it could simplify a lot of the setters if this was GetOrCreateObservation?

Comment (smacadam) [reply projects/VitalRecord/VitalRecord_util.cs:100]: changed name and updated setters

File: projects/VitalRecord/VitalRecord_util.cs (hunk 3 above)
Comment (smacadam) [reply projects/VitalRecord/VitalRecord_util.cs]: removed

File: projects/VitalRecord/VitalRecord_util.cs (hunk 4 above)
Comment (smacadam) [reply projects/VitalRecord/VitalRecord_util.cs]: added

**Pull request merged**